src/Avalonia.Styling/Styling/Style.cs

```diff
@@ -224,44 +147,12 @@ void ISetResourceParent.SetParent(IResourceNode parent)
 
             if (parent == null)
             {
-                Detach();
+                //Detach();
```

Comment:
Why is this commented out?

Reply:
Good point, and I forgot to mention it in the description. This PR actually removes support for removing styles (well, you can still remove them but they won't get unapplied to controls). This is because I wanted to implement _adding_ styles too and intended to do it in another PR. Not sure if this is a blocker for this PR.

Reply:
Does it have any observable difference if we don’t support it for now?

Reply:
If someone is removing styles from a `Styles` collection, yes. However the fact that _adding_ styles to these doesn't currently work in my mind makes it unlikely that anyone is relying on this. I plan to submit a PR to implement adding/removing styles in the next day or so anyway.